### PR TITLE
spyking circus containerized sorting wrapper

### PR DESCRIPTION
**Pull request by magland**

### CODE REVIEW

**Comment by magland** [comment]
@jsoules This is branched off of the slurm branch - so that PR should be merged first.

You should be able to just run the example sorting in: devel/sorting/test_sorting.py

File: devel/sorting/test_sorting.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            'recording_object': recording.object()
+        }).wait().return_value
+        sorting = le.LabboxEphysSortingExtractor(sorting_object)
+
+    unit_ids = sorting.get_unit_ids()
+    spike_train = sorting.get_unit_spike_train(unit_id=unit_ids[0])
+    print(f'Unit {unit_ids[0]} has {len(spike_train)} events')
+
+if __name__ == '__main__':
+    main()
```
Comment (magland) [inline devel/sorting/test_sorting.py]: Here's the example script for running containerized spyking circus.

File: spikeforest/sorters/spykingcircus/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+RUN apt-get update && apt-get install -y libglib2.0-0 libgl1-mesa-glx
+RUN apt-get update && apt-get install -y packagekit-gtk3-module libcanberra-gtk-module libcanberra-gtk3-module
+RUN pip install scikit-build
+RUN pip install cmake>=3.18
+RUN pip install spyking-circus==1.0.7
+
+# spikeinterface/spikesorters
+RUN pip install spikesorters==0.4.4
+
+LABEL version="1.0.7"
```
Comment (magland) [inline spikeforest/sorters/spykingcircus/docker/Dockerfile:19]: The Dockerfile recipe for defining the spyking circus image

File: spikeforest/sorters/spykingcircus/spykingcircus_wrapper1.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            auto_merge=auto_merge,
+            num_workers=num_workers,
+            whitening_max_elts=whitening_max_elts,
+            clustering_max_elts=clustering_max_elts
+        )     
+        timer = sorter.run()
+        print('#SF-SORTER-RUNTIME#{:.3f}#'.format(timer))
+        sorting = sorter.get_result()
+
+        return le.LabboxEphysSortingExtractor.store_sorting(sorting=sorting)
```
Comment (magland) [inline spikeforest/sorters/spykingcircus/spykingcircus_wrapper1.py]: The hither wrapper for spyking circus sorter

File: spikeforest/sorters/spykingcircus/spykingcircus_wrapper1.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        sorter.set_params(
+            detect_sign=detect_sign,
+            adjacency_radius=adjacency_radius,
+            detect_threshold=detect_threshold,
+            template_width_ms=template_width_ms,
+            filter=filter,
+            merge_spikes=merge_spikes,
+            auto_merge=auto_merge,
+            num_workers=num_workers,
```
Comment (jsoules) [inline spikeforest/sorters/spykingcircus/spykingcircus_wrapper1.py:49]: Hard-coded to 1: is this a requirement of the environment or just for proof-of-concept?

**Review by jsoules — COMMENTED** [review]
Only one comment from review, but I suppose I should probably try to run it before we merge?

**Pull request merged**